library/src/main/java/com/novoda/downloadmanager/DownloadBatch.java

```diff
@@ -50,6 +53,13 @@ void download() {
             return;
         }
 
+        if(!downloadConnectionAllowedChecker.isAllowedToDownload()) {
```

Comment:
How does this work if you `resume` a download? It's using the same instance right?

Reply:
Mmm can you describe the steps / expected result?

Reply:
- Start downloading
- Pause a download
- Trigger the wifi check thing (submitAllStoredDownloads)

Does it resume? 

Then try:

- Start downloading
- Pause a download
- Trigger the wifi check thing (submitAllStoredDownloads)
- Resume the download where the download now violates the network type check

Does it resume anyway?

Reply:
Ok, I just checked the two scenarios:
in the first one it doesn't resume only toggling the allowed connection, it stays paused.
In the second one when resuming after changing the condition to a violating one, it immediately fail for network error (restored later if the condition changes)